### ISSUE TITLE
Add Required Labels GitHub Action?

### ISSUE THREAD

**Issue by mathomp4** [GEOS-ESM/GEOSldas#369]
A query for @gmao-rreichle: Would you like me to add the Required Labels GitHub action to this repo that is in use in GEOSgcm, etc? Essentially it means no PR can be approved unless it has a "0 diff", etc. label. And will block any "contingent" label. I noticed you had the labels, but no action. Of course, the fact that not many people can merge into `develop` is, in its own way, a check against that! 😄

**Comment by gmao-rreichle** [comment]
Thanks for asking, @mathomp4 .  We didn't yet feel the need to have the labels enforced at the time of PR approval.  As you said, we're a small team and merging PRs is mostly done by committee (@weiyuan-jiang , @biljanaorescanin and myself).  But I suppose it couldn't hurt so we don't accidentally forget or inadvertently remove a label during the review process.  It would also help because it'll give developers a not-so-subtle hint that something is amiss if they create the PR without a label that indicates whether the PR is 0-diff or not. 
How does the "required labels github action" work in practice?  Does github enforce the label requirement by looking for strings? E.g., github would happy as long as one of the following labels is affixed: "0-diff", "0-diff trivial", or "not 0-diff", right?
Also, would this just be for PRs or also for Issues?  It wouldn't make much sense for the present "question" Issue...

**Comment by mathomp4** [comment]
@gmao-rreichle I actually misspoke. It's technically a webhook:

https://github.com/dimagi/required-labels

I found it before GitHub Actions. 

Essentially, we currently have `BANNED_LABELS` set to `Contingent - DNA,Needs Lead Approval` and `REQUIRED_LABELS_ANY` set to `0 diff,0 diff trivial,Non 0-diff,0 diff structural`. I could add your more verbose contingent label as well. Without any of the `REQUIRED_LABELS_ANY` or with one of the `BANNED_LABELS`, GitHub will not allow a branch to be merged. And we only use it on Pull Requests in GEOS. 

Note that this would also let me (if desired) to enable the "Update branch" button you've seen in places like the GEOSgcm_GridComp where it requires branches to be up-to-date before merging. GitHub only allows that function if there is a status check in place.

**Comment by gmao-rreichle** [comment]
> this would also let me (if desired) to enable the "Update branch" button
Thanks, @mathomp4 , now we're talking!  Yes, that'd be nice to have. 
The only downside that I can see is that the labels appear to be handmade for each repository.  I think I created most of them based on what I saw in other repos, and added a few to my own taste.  Also, all developers can easily edit the labels (or inadvertently remove them...), so there'll perhaps be a bit of maintenance involved. 
All told, though, let's give it a try, and throw in the "update branch" feature as an extra benefit.

**Comment by mathomp4** [comment]
@gmao-rreichle Okay. I think it is enabled now. It seems a bit twitchy, but I was turning labels on and off pretty quickly and it is a free service so...maybe I was doing something wonky.

And, if you look at the two "real" PRs here, you'll now see the "Update branch" buttons.

NOTE: make sure that the person developing the branch knows that button is pressed if they are still working on it. If you press "Update branch" and they don't pull from github, when they try to push, it'll conflict because their local history is now out-of-sync

**Comment by gmao-rreichle** [comment]
Many thanks, @mathomp4 !
We'll report back on how things work out.  I was thinking of the "update branch" button as a convenient tool mostly during the period when the developer is done with their part and we're all just waiting for something else to happen before we can proceed with the merge.  PRs often spend a good amount of time in this "holding" pattern. 
I'll do my best to spread the message among the land group. 
Thanks again!

**Comment by gmao-rreichle** [comment]
Successfully implemented by @mathomp4 , thanks!
